### PR TITLE
v2.2.0

### LINKED ISSUE #62: v2.2.0
v2.2.0 released: https://github.com/streamlink/streamlink/releases/tag/2.2.0

### PR DESCRIPTION
**Pull request by Conjuro**
- [x] v2.2.0 of Streamlink
- [x] Updated checksum
- [x] Updated link to changelog
- [x] Resolves #62
- [x] Pending package approval: https://chocolatey.org/packages/streamlink/2.2.0

### CODE REVIEW

File: Streamlink/tools/chocolateyinstall.ps1 [hunk 1]
```diff
@@ -13,11 +13,15 @@ $packageArgs = @{
...
-  checksum      = '3c27d90cfe1dd862cc56538e97efc686c2b9ece431fda3d6cd81163284ac2016'
+  checksum      = 'ca6b9664ae56892ae6a003bac8778a1458978e6457411ca4d17e2da26c2a0b53'
   checksumType  = 'sha256'
 
   silentArgs   = '/S'
   validExitCodes= @(0)
 }
 
+# v2.2.0: Changed default config file path (old path deprecated)
+Get-ChildItem -Path "$env:APPDATA\streamlink" streamlinkrc.* | Rename-Item -NewName { $_.Name -replace 'streamlinkrc','config' } -ErrorAction SilentlyContinue
```
Comment (Conjuro) [inline Streamlink/tools/chocolateyinstall.ps1]: Default config file paths changed and old path deprecated.

Added a line to rename. Will rename "streamlinkrc" and "streamlinkrc.*" (plugin configs) to "config" and "config.\*". If destination file exists, will silently error and continue, nothing gets overwritten.

**Review by bastimeyer — COMMENTED** [review]
Packages should never touch any user data, even on Windows. It's up to the user to fix that (unfortunately).

The problem here is that the package is using the Windows installer as source, and the installer is a little bit awkward with the config files, as it needs to create a default config with the `ffmpeg-ffmpeg` and `rtmp-rtmpdump` config vars set, which are required on Windows due to the lack of a sensible `PATH` env var for dependencies.

**Comment by Conjuro** [comment]
Yeah, that's why I wanted to check how you wanted it handled before I pushed anything out. I'll update the PR and roll a normal upgrade.

How'll it work out in the future when the deprecated file paths gets pulled from the code and the user hasn't moved their files? A clean install will setup the config as needed, but what about an update, will it also create the 'config' file?

**Comment by bastimeyer** [comment]
> How'll it work out in the future when the deprecated file paths gets pulled from the code and the user hasn't moved their files?

`%APPDATA%\streamlink\streamlinkrc` simply won't be read by default. And it won't do it right now when using the Windows installer, because it automatically creates `%APPDATA%\streamlink\config` and Streamlink will only read the first default config file it finds.

> what about an update, will it also create the 'config' file?

Since the package is using the Windows installer, it'll always create the `config` file when it doesn't exist.

- https://github.com/streamlink/streamlink/commit/b1c3aa14e867b440d926c95e94dc4d6fb6b2b0d4#diff-24a9b73e1130bf3bed74fa811baeb53c70bf59bc7be512c434d71e93c6c7f808
- https://github.com/streamlink/streamlink/blame/5b0687224fe9d44cd7c433c06e58d142bea8e8fa/script/makeinstaller.sh#L223-L230

**Comment by Conjuro** [comment]
> `%APPDATA%\streamlink\streamlinkrc` simply won't be read by default. And it won't do it right now when using the Windows installer, because it automatically creates `%APPDATA%\streamlink\config` and Streamlink will only read the first default config file it finds.

Gotcha. I'll put a comment on the package page (comment section) mentioning it, I foresee a few questions in the future from end-users wondering what's up.

**Comment by bastimeyer** [comment]
@gravyboat Has added a sticky thread on the streamlink issue tracker:
https://github.com/streamlink/streamlink/issues/3802

**Pull request merged**